core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,59 @@
+package tc.oc.pgm.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.math.Formula;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<StatType.OfFormula> formulaStats;
+
+  private StatsModule(List<StatType.OfFormula> formulaStats) {
+    this.formulaStats = formulaStats;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new StatsMatchModule(match, formulaStats);
+  }
+
+  public static class Factory implements MapModuleFactory<StatsModule> {
+    @Override
+    public StatsModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      XMLFluentParser parser = factory.getParser();
+      Element statsElement = XMLUtils.getUniqueChild(doc.getRootElement(), "stats");
+      if (statsElement == null) {
+        return new StatsModule(Collections.emptyList());
+      } else {
+        List<StatType.OfFormula> formulaStats = new ArrayList<>();
+        for (Element el : XMLUtils.getChildren(statsElement, "stat")) {
+          Component displayName = XMLUtils.parseFormattedText(el, "display-name");
+          double initialStatValue =
+              XMLUtils.parseNumber(el.getAttribute("initial"), Double.class, 0D);
+          if (displayName == null) {
+            throw new InvalidXMLException("Expected a display name to be present", el);
```

Comment:
You can (and should) the fluent parser for everything, it makes parsing less verbose.
Ie:
`parser.component(el, "name").required()` 
And just add support for component in there (that makes the call go parseFormattedText

Reply:
Cool can change